core/src/main/scala/dagr/core/execsystem/TopLikeStatusReporter.scala

```diff
@@ -52,32 +58,64 @@ object Terminal {
   */
 object CursorMovement {
   def positionCursor(line: Int, column: Int): String = s"\033[$line;${column}H"
-
   def moveUp(numLines: Int): String = s"\033[${numLines}A"
-
   def moveDown(numLines: Int): String = s"\033[${numLines}B"
-
   def moveRight(numColumns: Int): String = s"\033[${numColumns}C"
-
   def moveLeft(numColumns: Int): String = s"\033[${numColumns}D"
-
   def clearScreen: String = "\033[2J"
-
   def eraseToEOL: String = "\033[K"
+}
 
+object TopLikeStatusReporter {
+  /** Refreshes the top-like interface until terminate is called.
+    *
+    * @param reporter the reporter which to refresh
+    * @param refreshRate the refresh rate in ms.
+    */
+  private class StatusRunnable(reporter: TopLikeStatusReporter, refreshRate: Int = 1000) extends Runnable {
+    private val keepPrinting: AtomicBoolean = new AtomicBoolean(true)
+    def terminate(): Unit = keepPrinting.set(false)
+    override def run(): Unit = {
+      // skip over any current output
+      print(CursorMovement.clearScreen)
+      // keep printing until told otherwise
+      while (keepPrinting.get()) {
+        reporter.refresh(print)
+        Thread.sleep(refreshRate)
+      }
+    }
+  }
+  
+  object ReportField extends Enumeration {
+    type ReportField = Value
+    val ID, NAME, STATUS, STATE, CORES, MEMORY,
+    SUBMISSION_DATE, START_DATE, END_DATE,
+    EXECUTION_TIME, TOTAL_TIME, ATTEMPT_INDEX
+    = Value
+  }
 }
 
 /** Mix this trait into a [[dagr.core.execsystem.TaskManager]] to provide the `refresh()` method for a
   * top-like status interface. */
-trait TopLikeStatusReporter {
-  this: TaskManager =>
+class TopLikeStatusReporter(taskManager: TaskManager) {
+  import TopLikeStatusReporter.ReportField._
+
+  private val terminalRunnable = new StatusRunnable(this)
+  private val terminalThread = new Thread(terminalRunnable)
 
-  /** The header for the report */
-  private def reportHeader: List[String] = List(
-    "ID", "NAME", "STATUS", "STATE", "CORES", "MEMORY",
-    "SUBMISSION_DATE", "START_DATE", "END_DATE",
-    "EXECUTION_TIME", "TOTAL_TIME", "ATTEMPT_INDEX"
-  )
+  /** Start the top like status thread */
+  def start(): Unit = {
+    terminalThread.setDaemon(true)
+    terminalThread.start()
+  }
+
+  /** Shutdown the top like status thread */
+  def shutdown(): Unit = {
+    terminalRunnable.terminate()
+    terminalThread.join()
```

Comment:
Maybe add a timeout here so that if for some bizarre reason the thread is wedged it doesn't prevent shutdown?